aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -471,10 +471,11 @@ Tensor& tril_cpu_(Tensor &self, int64_t k) {
   if (self.numel() == 0) {
     return self;
   }
-  if (!checkTrilTriuBatchContiguous(self)) self = self.contiguous();
+  Tensor result = at::empty_like(self);
   AT_DISPATCH_ALL_TYPES(self.type(), "tril", [&]{
-    apply_triu_tril<scalar_t, true, false>(self, self, k);
+    apply_triu_tril<scalar_t, false, false>(result, checkTrilTriuBatchContiguous(self) ? self : self.contiguous(), k);
   });
+  self.resize_as_(result).copy_(result);
```

Comment:
you don't need the resize_as right?  Because the size is guaranteed to be correct.